clients/lightrider/lightrider-ui-features.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/* global ReportUIFeatures */
+
+/**
+ * Extends ReportUIFeatures to use the saved report from a browserified ReportGenerator.
+ */
+class LightriderUIFeatures extends ReportUIFeatures {
+  /**
+   * Uses ReportGenerator to create the html that recreates this report.
+   * @return {string}
+   * @override
+   */
+  getReportHtml() {
+    // @ts-ignore This is only called in browser in Lightrider where this exists.
```

Comment:
I'm not sure I understand this comment. It's called in the browser as in the WRS devtools script? Or it's called in the browser as-in PSI front-end?

Reply:
It's called in the browser as-in something similar to the PSI front-end.  

This ui-features is packaged with a report-generator for anyone trying to render a report internally on a front-end.  In order to do that they need this overrided getReportHtml since we can't just outerHTML.

Reply:
Ahhhhhh OK, the "in Lightrider" made that a bit confusing for me :)

this is happening in the consumers of the LR response just like in DevTools